syntax = "proto3";

package todo;

option py_generic_services = true;

service TodoService {
    rpc Create(CreateRequest) returns (CreateResponse);
    rpc Read(ReadRequest) returns (ReadResponse);
    rpc Update(UpdateRequest) returns (UpdateResponse);
    rpc Delete(DeleteRequest) returns (DeleteResponse);
    rpc List(ListRequest) returns (ListResponse);
}

message Todo {
    string id = 1;
    string title = 2;
}

message CreateRequest {
    string title = 1;
}

message CreateResponse {
    string message = 1;
}

message ReadRequest {
    string id = 1;
}

message ReadResponse {
    Todo todo = 1;
}

message UpdateRequest {
    string id = 1;
    string title = 2;
}

message UpdateResponse {
    string message = 1;
}

message DeleteRequest {
    string id = 1;
}

message DeleteResponse {
    string message = 1;
}

message ListRequest {}

message ListResponse {
    repeated Todo todos = 1;
}
